### PR TITLE
Fix spy infiltration cursor giving away fake buildings.

### LINKED ISSUE #10355: Fake structures revealed by spies
Hello everyone :) 

I just tested out (release 151224) : spies can't infiltrate fake structures (WF, Radar, Navy), so they can easily identify them.

I think ScottNZ talked about it here : https://www.bountysource.com/issues/7983378-add-more-fake-buildings

I think it would make the french faction even more interesting if Fake buildings could be infiltrable, without giving the veterancy (or reveal for Radar) as supposed . 

Is this behaviour planned for future releases ? 

Thank you all ! :)

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29502403-fake-structures-revealed-by-spies?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Fixes #10355.

This is a simple fix for an exploitable bug, so I'd like to see this included in the next playtest.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithDecoration.cs [hunk 1]
```diff
@@ -69,7 +69,17 @@ public WithDecoration(Actor self, WithDecorationInfo info)
 			Anim.PlayRepeating(info.Sequence);
 		}
 
-		public virtual bool ShouldRender(Actor self) { return true; }
+		protected virtual bool ShouldRender(Actor self)
+		{
+			if (self.World.RenderPlayer != null)
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Render/WithDecoration.cs]: Couldn't this be boiled down to:

`return self.World.RenderPlayer != null && Info.ValidStances.HasStance(self.Owner.Stances[self.World.RenderPlayer]);`

Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Render/WithDecoration.cs]: Well... looking at this again I see you are just moving existing code, so ignore this if you'd like.

**Comment by Phrohdoh** [comment]
This is odd...

```
Testing map: 02:   Five to one

OpenRA.Utility(1,1): Error: The map allows 1 possible players, but defines only 0 spawn points

[...]

Testing map: Survival 01

OpenRA.Utility(1,1): Error: Actor afld.mission is not defined by any rule.

OpenRA.Utility(1,1): Error: Actor afld.mission is not defined by any rule.

OpenRA.Utility(1,1): Error: Actor afld.mission is not defined by any rule.

OpenRA.Utility(1,1): Error: The map allows 1 possible players, but defines only 0 spawn points

[...]

Errors: 5

make: *** [test] Error 1
```

**Comment by pchote** [comment]
#12130 was hiding the real issue, which was:
```
System.AggregateException: One or more errors occurred. ---> System.ArgumentException: MiniYaml.Merge, duplicate values found for the following keys: Inherits: [Inherits (at ra|rules/fakes.yaml:29),Inherits (at ra|rules/fakes.yaml:30)]
```

**Comment by pchote** [comment]
Fixed.

**Review by obrakmann — APPROVED** [review]
lgtm, 👍

**Review by cjshmyr — APPROVED** [review]

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/aacc50311d2d87177102e6acdcefba63c587a3f6)

**Pull request merged**